clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -121,21 +122,24 @@ record_filter_t::parallel_shard_init_stream(int shard_index, void *worker_data,
             success_ = false;
         }
     }
+    std::lock_guard<std::mutex> guard(shard_map_mutex_);
```

Comment:
(Hmm, forgot that the tools are all using locks on parallel init -- makes the lock on exit not seem so bad.)

Reply:
Tracking shards is helpful as the tool can now produce per-shard results.